pkg/syncer/metadatasyncer.go

```diff
@@ -369,20 +369,41 @@ func pvDeleted(obj interface{}, metadataSyncer *MetadataSyncInformer) {
 		return
 	}
 	var deleteDisk bool
-	if pv.Spec.ClaimRef != nil && (pv.Status.Phase == v1.VolumeAvailable || pv.Status.Phase == v1.VolumeReleased) && pv.Spec.PersistentVolumeReclaimPolicy == v1.PersistentVolumeReclaimDelete {
-		klog.V(3).Infof("PVDeleted: Volume deletion will be handled by Controller")
+	if pv.Spec.PersistentVolumeReclaimPolicy == v1.PersistentVolumeReclaimDelete {
+		if pv.Spec.ClaimRef != nil {
+			if pv.Status.Phase == v1.VolumeAvailable || pv.Status.Phase == v1.VolumeReleased {
+				var metadataList []cnstypes.BaseCnsEntityMetadata
+				pvMetadata := cnsvsphere.GetCnsKubernetesEntityMetaData(pv.Name, nil, true, string(cnstypes.CnsKubernetesEntityTypePV), pv.Namespace)
```

Comment:
Regardless of any condition, all we need to do is just attempt to delete PV metadata when we receive call back that PV is deleted from Kubernetes. No need to check any condition like a phase or reclaim policy, as at the time we received this call back means PV is deleted from the Kubernetes cluster.

if update metadata is called after volume is deleted by the controller, we will get not found error we can just print that in the debug log.